frontend/src/app/components/containers/complaints/details/complaint-details-edit.tsx

```diff
@@ -249,7 +249,7 @@ export const ComplaintDetailsEdit: FC = () => {
       return;
     }
     if (hasValidationErrors()) {
-      dispatch(updateComplaintById(complaintUpdate, complaintType));
+      await dispatch(updateComplaintById(complaintUpdate, complaintType));
```

Comment:
Need the await here, otherwise getComplaint fires too early and the new values dont populate the complaint details page as it switches back to read only mode.

Reply:
wow egg on face!
 
It goes against my understanding and experience, but it seems like this does work when using "redux-thunk" middleware which is part of redux toolkit: https://www.reddit.com/r/reduxjs/comments/zuuv04/await_dispatch_async_actionredux_tookitcan_anyone/ 
 
If not using thunks, the standard redux pattern of raising downstream actions would be required to get the chained behaviour.